extractor/text.go

```diff
@@ -1180,17 +1181,46 @@ var (
 // sortPosition sorts a text list by its elements' positions on a page.
 // Sorting is by orientation then top to bottom, left to right when page is orientated so that text
 // is horizontal.
+// Text is considered to be on different lines if the lines' orientedStart.Y differs by more than `tol`.
 func (pt *PageText) sortPosition(tol float64) {
+	if len(pt.marks) == 0 {
+		return
+	}
 	sort.SliceStable(pt.marks, func(i, j int) bool {
 		ti, tj := pt.marks[i], pt.marks[j]
 		if ti.orient != tj.orient {
 			return ti.orient < tj.orient
 		}
-		if math.Abs(ti.orientedStart.Y-tj.orientedStart.Y) > tol {
+		if ti.orientedStart.Y != tj.orientedStart.Y {
 			return ti.orientedStart.Y > tj.orientedStart.Y
```

Comment:
Should it be >= ?  Given that the function for SliceStable is "less" ?  Is this ascending sort?

Reply:
This code has been removed in my simplification